web/src/main/java/com/bazaarvoice/emodb/web/scanner/ScanUploader.java

```diff
@@ -190,18 +197,20 @@ private ScanStatus createNewScanFromExistingScan(String scanId, ScanOptions scan
     }
 
     private void startScanUpload(String scanId, ScanStatus status, Set<String> placements) {
-        try {
-            // compaction control timestamp = stash start time + 1 minute buffer time. This is needed to allow the setting time to trickle the request to the DataStore.
-            // Setting the time in the future takes care of the issue of there being any in-flight compactions
-            // Note: the same compaction control timestamp with 1 minute buffer time is also considered during the multiscan deltas/compactions resolving.
-            long compactionControlTime = status.getCompactionControlTime().getTime();
-            // expired time for now is designed to be 10 hours from the compaction control time.
-            long expireTime = compactionControlTime + Duration.ofHours(10).toMillis();
-            // Update the scan start time in Zookeeper in all data centers.
-            _compactionControlSource.updateStashTime(scanId, compactionControlTime, Lists.newArrayList(status.getOptions().getPlacements()), expireTime, _dataCenters.getSelf().getName());
-        } catch (Exception e) {
-            _log.error("Failed to update the stash time for scan {}", scanId, e);
-            throw Throwables.propagate(e);
+        if (status.getOptions().isTemporalEnabled()) {
```

Comment:
I was curious if we need to add this check at the stash time deletion places as well. But looks like Zookeeper MapStore is fine with deletions of non-existent entries.

Reply:
Exactly, I looked into it, and it seemed like it was better for the delete to just remain a no-op.